From 2939bb514c72e3059ee258e9c3e99716ae0f9066 Mon Sep 17 00:00:00 2001
From: Jonas Larsson <ljonas@google.com>
Date: Mon, 26 Aug 2019 15:31:08 -0700
Subject: [PATCH 08/17] gldownload: download RGB, BGR and BGRA with dmabuf

Allow these formats to be downloaded using ION backed dmabufs,
which is orders of magnitude faster than PBO transfers.

This change depends on an updated imx-gpu-viv and allows for GPU
based conversions from YUV (camera, VPU) to inference friendly
RGB (not RGBA) at the native source rate (30 fps).

Change-Id: I04c3ca9261d9ba911dff14c0df9b996637b481fb
---
 ext/gl/gstgldownloadelement.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/ext/gl/gstgldownloadelement.c b/ext/gl/gstgldownloadelement.c
index 8b4ea4ba4..d3c229085 100644
--- a/ext/gl/gstgldownloadelement.c
+++ b/ext/gl/gstgldownloadelement.c
@@ -518,8 +518,11 @@ gst_gl_download_element_propose_allocation (GstBaseTransform * bt,
 
   GST_DEBUG_OBJECT (bt, "video format is %s", gst_video_format_to_string (fmt));
 
-#if GST_GL_HAVE_IONDMA
-  if (fmt == GST_VIDEO_FORMAT_RGBA || fmt == GST_VIDEO_FORMAT_RGB16) {
+  #if GST_GL_HAVE_IONDMA
+  if (fmt == GST_VIDEO_FORMAT_RGBA || fmt == GST_VIDEO_FORMAT_RGB16 ||
+      fmt == GST_VIDEO_FORMAT_RGB ||
+      fmt == GST_VIDEO_FORMAT_BGRA ||
+      fmt == GST_VIDEO_FORMAT_BGR ) {
     allocator = gst_gl_memory_dma_allocator_obtain ();
     GST_DEBUG_OBJECT (bt, "obtain dma memory allocator %p.", allocator);
   }
-- 
2.26.0

